syntax = "proto3";

package communication_objects;

enum ComponentType {
    META = 0;
    LEARNER = 1;
    MULTIENV = 2;
    ENVIRONMENT = 3;
    EVALUATION = 4;
    AGENTS = 5;
}

enum StatusType {
    ERROR = 0;
    // module is still "launching" or waiting something
    LAUNCHING = 1;
    // ready to run or running
    RUN = 2;
}

enum EnvironmentCommand {
    STEP = 0;
    RESET = 1;
    CLOSE = 2;
}

enum LoadType {
    LOCAL = 0;
    AWS = 1;
}